ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/AbstractCreateStreamCommand.java

```diff
@@ -165,64 +138,26 @@ static void checkMetaData(
   }
 
   private RegisterTopicCommand registerTopicFirst(
-      final Map<String, Literal> properties
+      final StatementProperties properties
   ) {
-    final Literal topicNameExp = properties.get(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY);
-
-    if (topicNameExp == null) {
-      throw new KsqlException("Corresponding Kafka topic ("
-          + DdlConfig.KAFKA_TOPIC_NAME_PROPERTY + ") should be set in WITH clause.");
-    }
-
-    final String kafkaTopicName = StringUtil.cleanQuotes(topicNameExp.toString());
+    final String kafkaTopicName = StringUtil.cleanQuotes(properties.getKafkaTopic());
```

Comment:
`StringUtil.cleanQuotes` could be done inside the `getKafkaTopic` call

Reply:
ah yes! it is being done :)